### PR TITLE
Add EmptyOperator and decorator to decorators/empty.py

### PR DESCRIPTION
**Pull request by AlefRP**
# Add EmptyOperator with @task.empty decorator

## Objective
This PR introduces a new operator called `EmptyOperator` with support for the `@task.empty` decorator. This operator serves as a placeholder for tasks in DAGs that do not have any execution logic but are useful for defining the structure of the workflow.

## Key Changes
- Created `airflow/decorators/empty.py` with the implementation of `EmptyOperator` and the `@task.empty` decorator.
- Modified `airflow/decorators/__init__.pyi` to expose the new `empty` decorator.
- Added functionality to capture `args/kwargs` and log custom messages when executing tasks decorated with `@task.empty`.

## Example Usage
```python
from airflow.decorators import dag, task
from airflow.utils.dates import days_ago

@dag(schedule_interval=None, start_date=days_ago(1), catchup=False)
def my_empty_dag():
    @task.empty
    def start_task():
        print("This is a placeholder start task.")

    @task.empty
    def end_task():
        print("This is a placeholder end task.")

    start_task() >> end_task()

dag = my_empty_dag()

### CODE REVIEW

**Comment by eladkal** [comment]
Given that we plan to move decorators to standard provider https://github.com/apache/airflow/pull/44027 it's best to add new decorators to the standard provider directly rather than to the old path

**Comment by potiuk** [comment]
I think it's also not a good idea. The whole idea of an EmptyOperator is that it is empty. so decorating a method with it makes no sense - because it will never have a callable. The whole idea of an empty operator is that it is there, serviing as a placeholder but it is also skipped when scheduling. Enttirely skipped in scheduler without even attempting to run anything - without even considering running anything.

As far as I understand it now - when it decorates a funtion, it's essentially the same as `@task` - the only difference is that it wraps around EmptyOperator, but - unless I understand it wrongly and please correct me if I am wrong - it will actually attempt to execute the "execute()" method (unlike when you use EmptyOperator in your DAG directly). This is also why you have logs at execution time.

So basically it's no different to write:

```
@task.empty()
def empty_task_a():
     pass
```

than

```
@task
def empty_task_a():
     pass
```

IMHO - they will behave exactly the same.

Or am I wrong @AlefRP ?
